packages/common/src/types/ccdAmount.ts

```diff
@@ -22,7 +49,83 @@ export class CcdAmount {
         this.microCcdAmount = microCcdAmount;
     }
 
+    /**
+     * Returns the amount of micro CCD as a Big.
+     *
+     * @returns The amount of micro CCD as a Big
+     */
+    toBig(): Big {
+        return Big(this.microCcdAmount.toString());
+    }
+
+    /**
+     * Returns the amount of CCD as a Big.
+     *
+     * @returns The amount of CCD as a Big
+     */
+    toCcd(): Big {
+        return this.toBig().div(Big(MICRO_CCD_PER_CCD));
+    }
+
+    /**
+     * Creates a CcdAmount from a number, string, big or bigint.
+     *
+     * @param ccd The amount of CCD
+     * @returns The CcdAmount object derived from the ccd input parameter
+     * @throws If the number of commas in the input string exceeds 1
+     * @throws If a number is passed with several decimal seperators
+     * @throws If a negative amount of micro CCD is passed
+     * @throws If the micro CCD passed is greater than what can be contained in a 64-bit integer
+     */
+    static fromCcd(ccd: BigSource | bigint): CcdAmount {
+        if (typeof ccd === 'bigint') {
+            ccd = ccd.toString();
+        }
+
+        const microCcd = newBig(ccd).mul(Big(MICRO_CCD_PER_CCD));
+        return new CcdAmount(microCcd);
+    }
+
+    /**
+     * Converts an amount of CCD to micro CCD and asserts that the amount is a valid amount of CCD.
+     *
+     * @param ccd The amount of CCD
+     * @returns The amount of micro CCD as a Big
+     * @throws If the number of commas in the input string exceeds 1
+     * @throws If a number is passed with several decimal seperators
+     */
+    static ccdToMicroCcd(ccd: BigSource | bigint): Big {
+        return CcdAmount.fromCcd(ccd).toBig();
+    }
+
+    /**
+     * Converts an amount of micro CCD to CCD and asserts that the amount is a valid amount of CCD.
+     *
+     * @param microCcd The amount of micro CCD as a number, string, big or bigint.
+     * @returns The amount of CCD as a Big
+     * @throws If the number of commas in the input string exceeds 1
+     * @throws If a number is passed with several decimal seperators
+     * @throws If a negative amount of micro CCD is passed
+     * @throws If the micro CCD passed is greater than what can be contained in a 64-bit integer
+     */
+    static microCcdToCcd(microCcd: BigSource | bigint): Big {
+        return new CcdAmount(microCcd).toCcd();
+    }
+
     toJSON(): string {
         return this.microCcdAmount.toString();
     }
 }
+
+function newBig(bigSource: BigSource): Big {
+    if (typeof bigSource === 'string') {
+        const numberOfCommaSeperators = bigSource.split(',').length - 1;
+
+        if (numberOfCommaSeperators > 1) {
+            throw Error('More than one decimal seperator found!');
+        } else if (numberOfCommaSeperators === 1) {
+            return Big(bigSource.replace(',', '.'));
+        }
+    }
+    return Big(bigSource);
+}
```

Comment:
Here you only validate that numbers with comma separators have 0 or 1 separator, but is there validation in place for e.g. `12.0.2` in the `Big` constructor? And if so, would it not be easier to simplify this code to
```suggestion
function newBig(bigSource: BigSource): Big {
    if (typeof bigSource === 'string') {
        return Big(bigSource.replace(',', '.'));
    }
    return Big(bigSource);
}
```